react/src/components/Table/Table.tsx

```diff
@@ -592,6 +667,18 @@ const Table: React.FC<TableProps> = ({ variantData }) => {
                                 page.map(row => {
                                     prepareRow(row);
                                     const { key, ...restRowProps } = row.getRowProps();
+                                    const caseDetailsCol = headerGroups[0].headers.find(
+                                        header => header.Header === 'Case Details'
+                                    );
+                                    // Display only one row per variant if Case Details Section is collapsed.
+                                    if (caseDetailsCol && !isHeaderExpanded(caseDetailsCol)) {
+                                        if (
+                                            uniqueVariantIndices.find(i => i === row?.index) ===
```

Comment:
In our codebase, we often right `!uniqueVariantIndices.find(i => i === row?.index)` which is the equivalent of `uniqueVariantIndices.find(i => i === row?.index) === undefined`

Reply:
`!uniqueVariantIndices.find(i => i === row?.index)` would ignore rows that have `index=0`. 

`index` is a property for rows: [link]( https://react-table.tanstack.com/docs/api/useTable#row-properties). With the underlying implementation, the index starts with 0. It might be more robust to conserve the default implementation.

Reply:
Ah that's right. Agreed.